web/src/stream_ui_updates.js

```diff
@@ -228,22 +229,37 @@ export function update_add_subscriptions_elements(sub) {
     const $button_element = $add_subscribers_container
         .find('button[name="add_subscriber"]')
         .expectOne();
-    const allow_user_to_add_subs = sub.can_add_subscribers;
+
+    // We should disable the widgets when stream is private or
+    // realm level setting does not allow user to add subscribers.
+    const allow_user_to_add_subs =
+        sub.can_add_subscribers && settings_data.user_can_subscribe_other_users();
```

Comment:
I think this check of `user_can_subscribe_other_users()` should be added to `stream_data.can_subscribe_others` instead of adding it here.

Reply:
Do we want a common tooltip message for all cases? At present there are two different tooltip messages depending on whether adding subscribers was disabled because stream was private or because of realm level setting.

Reply:
What are the two messages?

Reply:
As per present implementation if stream is private and user is not subscribed to stream we show `Only stream members can add users to a private stream` and if user is not allowed to add subscribers because of realm level settings we show `You are not allowed to add subscribers to stream.`

Reply:
Hmm, I think it's helpful to keep those two separate -- the first one provides good info for org admins.

There should be a period after the first one, and let's change the second one to, "You are not allowed to add subscribers to this stream." for better grammar.